CRM/Contribute/BAO/Contribution.php

```diff
@@ -1833,7 +1833,22 @@ public static function transitionComponents($params) {
       }
     }
 
-    $contribution->loadRelatedObjects($input, $ids);
+    $paymentProcessorID = CRM_Utils_Array::value('payment_processor_id', $input, CRM_Utils_Array::value(
```

Comment:
We should be using ?? instead of CRM_Utils_Value but I can see this is a straight extraction so ok.

Reply:
@mattwire yeah - I avoided any cleanup in it